frame/support/procedural/src/pallet/expand/store_trait.rs

```diff
@@ -36,8 +36,19 @@ pub fn expand_store_trait(def: &mut Def) -> proc_macro2::TokenStream {
 	let storage_names = &def.storages.iter().map(|storage| &storage.ident).collect::<Vec<_>>();
 	let storage_cfg_attrs =
 		&def.storages.iter().map(|storage| &storage.cfg_attrs).collect::<Vec<_>>();
+	let warnig_struct_name = syn::Ident::new("Store", *attribute_span);
+	let warning: syn::ItemStruct = syn::parse_quote!(
+		#[deprecated(note = r"
+		Use of `#[pallet::generate_store(pub(super) trait Store)]` will be removed soon.
+		Check https://github.com/paritytech/substrate/pull/13535 for more details.")]
+		struct #warnig_struct_name;
+	);
 
 	quote::quote_spanned!(trait_store.span() =>
+		mod warnig {
```

Comment:
```suggestion
		const _ :() = {
```

This should also work? It would be better, because then there can be no clash with the name `warning`.

Reply:
It does not work as struct and trait both named as `Store`. So I think we should keep warning mod.

Reply:
Then give the struct a different name?

Reply:
is it really necessary? I mean hopefully this code will be removed eventually

Reply:
however if you still insist please suggest what name you prefer?

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0b1a74694278f0f8f9b39b14cdc6c793 not sure what error you have seen while trying this?

Reply:
Oh sorry, I did not misread your suggestion, I tried removing mod warning only and it gave me error.
your suggestion works fine.